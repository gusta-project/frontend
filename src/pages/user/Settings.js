import { Helmet } from 'react-helmet';
import React, { Component } from 'react';
import { Container, Row, Col, Form, InputGroup, Button } from 'react-bootstrap';

export default class UserSettings extends Component {
  constructor(...args) {
    super(...args);

    this.state = {
      username: 'xXTeddyBearSlayer69Xx',
      email: 'SlayTeddiesAllDay@aol.com',
      filename: '',
      frontPageList: [
        {
          id: 1,
          name: 'Top Recipes of the Day'
        },
        {
          id: 2,
          name: 'Newest Recipes'
        },
        {
          id: 3,
          name: 'Featured Mixer'
        }
      ],
      droppedItems: [],
      draggedItem: {}
    };
  }

  handleFileInput(event) {
    this.setState({ filename: event.target.files[0].name });
  }

  render() {
    return (
      <Container>
        <Helmet title="Your Settings" />
        <Row className="text-center">
          <Col>
            <h1>User Settings</h1>
          </Col>
        </Row>
        <Form>
          <Row className="text-center">
            <Col>
              <InputGroup>
                <InputGroup.Prepend>
                  <InputGroup.Text>Username</InputGroup.Text>
                </InputGroup.Prepend>
                <Form.Control
                  name="username"
                  type="text"
                  defaultValue={this.state.username}
                  placeholder={this.state.username}
                  aria-describedby="Username"
                />
              </InputGroup>
            </Col>
            <Col>
              <InputGroup>
                <InputGroup.Prepend>
                  <InputGroup.Text>E-mail</InputGroup.Text>
                </InputGroup.Prepend>
                <Form.Control
                  name="e-mail"
                  type="email"
                  defaultValue={this.state.email}
                  placeholder={this.state.email}
                  aria-describedby="E-mail address"
                />
              </InputGroup>
              <br />
            </Col>
          </Row>
          <Row className="text-center">
            <Col md="3" className="align-self-center">
              <img
                src="/media/harold.jpeg"
                alt="Current profile pic"
                className="w-50"
              />
            </Col>
            <Col md="6" className="align-self-center">
              <h3>Profile Picture</h3>
              <InputGroup>
                <div className="custom-file">
                  <Form.Control
                    name="profilePic"
                    type="file"
                    className="custom-file-input"
                    onChange={event => this.handleFileInput(event)}
                  />
                  <Form.Label className="custom-file-label text-left">
                    {this.state.filename
                      ? this.state.filename
                      : 'Choose a picture'}
                  </Form.Label>
                </div>
                <InputGroup.Prepend>
                  <Button className="button-animation">
                    <span>Upload</span>
                  </Button>
                </InputGroup.Prepend>
              </InputGroup>
            </Col>
          </Row>
          <hr />
          <Row className="text-center">
            <Col />
          </Row>
          <Row className="text-center">
            <Col>
              <Button className="button-animation">
                <span>Save</span>
              </Button>
            </Col>
          </Row>
        </Form>
      </Container>
    );
  }
}
