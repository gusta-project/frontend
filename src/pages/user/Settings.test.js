import React from 'react';
import renderer from 'react-test-renderer';
import { UserSettings } from './Settings';

describe('<UserSettings />', () => {
  const actions = {
    toggleDarkMode: jest.fn()
  };
  const props = {
    theme: 'default',
    actions
  };

  it('renders correctly', () => {
    expect(
      renderer.create(<UserSettings {...props} />).toJSON()
    ).toMatchSnapshot();
  });

  it('renders correctly with dark mode enabled', () => {
    expect(
      renderer.create(<UserSettings {...props} theme="dark" />).toJSON()
    ).toMatchSnapshot();
  });
});
