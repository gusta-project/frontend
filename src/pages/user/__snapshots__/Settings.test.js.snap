// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSettings /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="text-center row"
  >
    <div
      className="col"
    >
      <h1>
        User Settings
      </h1>
    </div>
  </div>
  <form
    className=""
  >
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              Username
            </span>
          </div>
          <input
            aria-describedby="Username"
            className="form-control"
            defaultValue="xXTeddyBearSlayer69Xx"
            name="username"
            placeholder="xXTeddyBearSlayer69Xx"
            type="text"
          />
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              E-mail
            </span>
          </div>
          <input
            aria-describedby="E-mail address"
            className="form-control"
            defaultValue="SlayTeddiesAllDay@aol.com"
            name="e-mail"
            placeholder="SlayTeddiesAllDay@aol.com"
            type="email"
          />
        </div>
        <br />
      </div>
    </div>
    <div
      className="text-center row"
    >
      <div
        className="align-self-center col-md-3"
      >
        <img
          alt="Current profile pic"
          className="w-50"
          src="/media/harold.jpeg"
        />
      </div>
      <div
        className="align-self-center col-md-6"
      >
        <h3>
          Profile Picture
        </h3>
        <div
          className="input-group"
        >
          <div
            className="custom-file"
          >
            <input
              className="custom-file-input form-control-file"
              name="profilePic"
              onChange={[Function]}
              type="file"
            />
            <label
              className="custom-file-label text-left form-label"
            >
              Choose a picture
            </label>
          </div>
          <div
            className="input-group-prepend"
          >
            <button
              className="button-animation btn btn-primary"
              disabled={false}
              type="button"
            >
              <span>
                Upload
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="text-center row"
    >
      <div
        className="align-self-center col-md-12"
      >
        Toggle Dark Mode
        <label
          className="switch my-auto mx-1"
        >
          <input
            checked={false}
            name="Enable Dark Mode"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="slider slider--round"
          />
        </label>
      </div>
    </div>
    <hr />
    <div
      className="text-center row"
    >
      <div
        className="col"
      />
    </div>
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <button
          className="button-animation btn btn-primary"
          disabled={false}
          type="button"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`<UserSettings /> renders correctly with dark mode enabled 1`] = `
<div
  className="container"
>
  <div
    className="text-center row"
  >
    <div
      className="col"
    >
      <h1>
        User Settings
      </h1>
    </div>
  </div>
  <form
    className=""
  >
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              Username
            </span>
          </div>
          <input
            aria-describedby="Username"
            className="form-control"
            defaultValue="xXTeddyBearSlayer69Xx"
            name="username"
            placeholder="xXTeddyBearSlayer69Xx"
            type="text"
          />
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              E-mail
            </span>
          </div>
          <input
            aria-describedby="E-mail address"
            className="form-control"
            defaultValue="SlayTeddiesAllDay@aol.com"
            name="e-mail"
            placeholder="SlayTeddiesAllDay@aol.com"
            type="email"
          />
        </div>
        <br />
      </div>
    </div>
    <div
      className="text-center row"
    >
      <div
        className="align-self-center col-md-3"
      >
        <img
          alt="Current profile pic"
          className="w-50"
          src="/media/harold.jpeg"
        />
      </div>
      <div
        className="align-self-center col-md-6"
      >
        <h3>
          Profile Picture
        </h3>
        <div
          className="input-group"
        >
          <div
            className="custom-file"
          >
            <input
              className="custom-file-input form-control-file"
              name="profilePic"
              onChange={[Function]}
              type="file"
            />
            <label
              className="custom-file-label text-left form-label"
            >
              Choose a picture
            </label>
          </div>
          <div
            className="input-group-prepend"
          >
            <button
              className="button-animation btn btn-primary"
              disabled={false}
              type="button"
            >
              <span>
                Upload
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="text-center row"
    >
      <div
        className="align-self-center col-md-12"
      >
        Toggle Dark Mode
        <label
          className="switch my-auto mx-1"
        >
          <input
            checked={true}
            name="Disable Dark Mode"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="slider slider--round"
          />
        </label>
      </div>
    </div>
    <hr />
    <div
      className="text-center row"
    >
      <div
        className="col"
      />
    </div>
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <button
          className="button-animation btn btn-primary"
          disabled={false}
          type="button"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
